### DISCUSSION TITLE
Slideshow with older Browsers

### DISCUSSION THREAD

**Discussion by fabian1512** [damongolding/immich-kiosk#167] [Feature request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Unfortunately, it is currently not possible to display this fantastic slideshow on older browsers.
I would be delighted if the compatible browsers were extended to older versions.
I would be happy to do without optical functions if they would no longer work due to compatibility.
I think older devices in particular can still be given a reason for existence with the function of a digital picture frame.
In my case, I would like to breathe new life into an iPad 4 with iOS 10.

**Comment by damongolding** [comment]
Hey. 

It's always a tough decision where to draw the line in terms of compatibility. Like yourself, I recently tried to get Kiosk running on my TV but the maker has locked it to a quite old firmware. 

It would require a full rebuild to make Kiosk work on older browsers well, which I am unwilling to do. 

Having said that, I am willing to create a small legacy build pipeline. By that I mean I can use some automation tools to try and transpile Kiosk's JavaScript down to a more compatible version. 

I don't have an iPad so I am unable to test the outcome. So if you're willing I can ping you when it's ready for a test drive?

**Comment by fabian1512** [comment]
Hi,
that would of course be great if you could try this with little effort. And yes, I would love to test it

**Comment by damongolding** [comment]
@fabian1512 ok, so you can try this docker image `damongolding/immich-kiosk-development:0.12.2` with the url param `legacy=true` e.g. `http://KIOSK_URL:PORT?legacy=true`

**Reply by fabian1512** [reply]
Thank you very much for your efforts. This version runs perfectly on my M1 Mac, but unfortunately not on the old iPad. Here I only get a black background and the colourful star and then it stops.

**Reply by damongolding** [reply]
That's a shame, I assume it's the ajax engine failing. Are you in the Immich Discord server?

**Reply by fabian1512** [reply]
no, I am not

Now I am

**Reply by damongolding** [reply]
Can you visit [https://v1.htmx.org/test/](https://v1.htmx.org/test/) on you iPad for me? and report back the results?

**Reply by fabian1512** [reply]
sadly nothing else than that:
![IMG_0002](https://github.com/user-attachments/assets/67728bd2-a101-4ec9-a50f-8ab00fad80da)

**Reply by damongolding** [reply]
I pushed another image with some polyfills `damongolding/immich-kiosk-development:0.12.2-beta.1`, again the `legacy=true` will be needed.

**Reply by fabian1512** [reply]
same same 😢

**Reply by fabian1512** [reply]
I switched to another tablet, that can handle kiosk. So many thanks for your efforts so far.
Maybe you can keep my suggestion for revitalising old tablets in mind and find a solution sometime later

**Reply by damongolding** [reply]
> I switched to another tablet, that can handle kiosk. So many thanks for your efforts so far.
> 
> Maybe you can keep my suggestion for revitalising old tablets in mind and find a solution sometime later

I'd love to get Kioks working on old devices. Buts it's quite hard for me to debug old browsers as I don't have access to any.

**Reply by hctrr** [reply]
@fabian1512, a route I took is to provision a Linux desktop and RDP from the old device.
I used [linuxserver/docker-webtop](https://github.com/linuxserver/docker-webtop ) then launched a more modern browser, full screen it, and it fly.

@damongolding, let me know if you're still looking for testers.

**Reply by fabian1512** [reply]
@hctrr, thank you very much for your suggestion - sounds like a possible solution. I'll give it a try

**Comment by jo-gross** [comment]
I’m also running the kiosk on my newer devices. During the Christmas holidays, I found my old iPad (2nd gen) and would like to turn it into a digital photo frame. For that, I need the same functionality described here, but unfortunately it doesn’t work either.

I found another repository that targets this problem (in case anyone is interested): https://github.com/damarte/immich-slideshow-old-devices

Before I start trying to customize it for my own use (it’s currently very basic: fixed image sizes, not well rotated images, and no password - all in all a great base, but not enough for me and i don´t want to host 2 different projects), I wanted to check whether this topic is still on the roadmap for this project. I’d be happy to help with this. 

@damongolding if you’re interested, we could work on it together, for example in a call, to figure out how we might fix it (i could provide the old iPad, but have no idea about the project and its structure) — if it’s still of interest.

**Comment by aka-charos** [comment]
@jo-gross , did you try the image endpoint?  https://docs.immichkiosk.app/misc/image/ 
That's how I am running immich-kiosk on my old ios 10 ipad

**Reply by jo-gross** [reply]
Thanks for your input! Yeah, thats working, even if it’s not as complete as i wanted (like missing the clock, photo location, creation date or album).
I solved the problem for myself by programming a very light weight immich kiosk only for my old devices. I also added an overlay option for an album selection.

**Comment by damongolding** [comment]
I created a base concept for legacy devices here https://github.com/damongolding/kiosk-legacy-experiment. I don't have an old iPad etc but a user on Discord said it was working on Safari 9 (with some minor issues).

**Reply by fabian1512** [reply]
Sounds great - so I'll give it a try on my old iPad
Many thanks

**Reply by fabian1512** [reply]
@damongolding: I´ve tried the installation via docker. with http://kiosk-IP:4000 the page remains white, no pictures shown up.
Did I miss something during installation?

**Reply by damongolding** [reply]
Is that following the instructions on the readme?

The URL should be http://SERVER:4000 server meaning the machine the docker container is running. 

Also test in a desktop browser first to make sure the service is running. If it's still not working check the browsers console for any errors.

**Reply by fabian1512** [reply]
Great, got it running. Since the 0.33.0-cors dev build rejected the allowed_origins config (validation error), I worked around the CORS issue with an Nginx reverse proxy:

1. Created an Nginx container listening on port 3001, proxying to Kiosk on port 3000, injecting Access-Control-Allow-Origin headers
2. Set VITE_KIOSK_URL="http://<host>:3001" in .env
3. Rebuilt the legacy container with docker compose build --no-cache

nginx-cors.conf:
```
server {
    listen 3001;
    location / {
        proxy_pass http://<kiosk-host>:3000;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;

**Reply by damongolding** [reply]
The CORS config is in the main 
 build as on the v0.33.0 release https://docs.immichkiosk.app/configuration/additional-options/#cors, either way I'm glad you got it up and running 👍🏼

**Reply by fabian1512** [reply]
got it now. the part in the readme
kiosk:
  allowed_origins:
    - http://localhost:5432 # This is the default port for the development server
    # - http://localhost:4173 # This is the default port for the preview server

was a bit confusing. Just switched ist to "http://kiosk-ip:4000" and it worked

**Comment by Fenisu** [comment]
I just wanted to let you know that this commit on legacy repo:
https://github.com/damongolding/immich-kiosk-legacy/commit/b921e7246e221a7d6d4dfac6ce85d1a301e983a8

works on an iPad Air 2 with Safari 15.6.

**Reply by damongolding** [reply]
I have a user saying that the normal Kiosk version works for them on Safari 12.1 (#777) @Fenisu
